base/reducedim.jl

```diff
@@ -124,8 +105,33 @@ function _reducedim_init(f, op, fv, fop, A, region)
     return reducedim_initarray(A, region, z, Tr)
 end
 
-reducedim_init(f, op::typeof(max), A::AbstractArray{T}, region) where {T} = reducedim_initarray0(A, region, f, maximum)
-reducedim_init(f, op::typeof(min), A::AbstractArray{T}, region) where {T} = reducedim_initarray0(A, region, f, minimum)
+# initialization when computing minima and maxima requires a little care
+for (f1, f2, initval) in ((min, max, Inf), (max, min, -Inf))
+    function reducedim_init(f, op::typeof(f1), A::AbstractArray, region)
```

Comment:
This needs an `@eval` with interpolation — I don't think this will work as it stands.